---
title: "WS1 Introduction to Bioconductor 3.12"
author: Vince Carey^[stvjc@channing.harvard.edu]
output: 
  rmarkdown::html_vignette:
    fig_height: 4
    fig_width: 5
vignette: >
  %\VignetteIndexEntry{WS1 Introduction to Bioconductor 3.12}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
bibliography: biocwk312.bib
---


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
suppressPackageStartupMessages({
library(ggplot2)
library(BSgenome)
library(BSgenome.Scerevisiae.UCSC.sacCer3)
library(TxDb.Scerevisiae.UCSC.sacCer3.sgdGene)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(AnnotationHub)
})
```


# Prerequisites and motivating examples

## Pre-requisites

* Basic knowledge of R syntax
* Familiarity with Rstudio
* Basic understanding of modern genomics.  For example, the distinction
between whole genome sequencing and RNA-seq should be clear.
* Basic familiarity with the concepts of statistical analysis, such
as the definition of the t-test for comparing sample means, the
interpretation of histograms.  An understanding of experimental design
is helpful.
* Optional readings:
    - [An introduction to R](https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf)
    - [The beginners page](https://education.rstudio.com/learn/beginner/) at the Rstudio education portal.
    - [An introduction to Bioconductor](https://www.datacamp.com/community/tutorials/intro-bioconductor) at DataCamp.


## Programming and package use with R 4.0.3

R is a programming language focusing on activities
important in statistical data analysis, but it is
often thought of more as an interactive environment
for working with data.  This concept of "environment"
will become clearer as we use R to perform progressively
mode complex tasks.

### Simple illustrations

- simple arithmetic
```{r ill1}
(7+5)/(3*6)
```
- assignment; string definitions and operations
```{r ill2}
s = "abc"
s
paste(s, s, sep="")
```
- random number generation
```{r ill3}
set.seed(1234) # setting a seed for reproducibility
x = rnorm(5)
x
```
- data visualization
```{r ill4}
y = rnorm(5)
plot(x,y)
```

### Making new variables

In the example `s = "abc"`, we formed a _string constant_
by enclosing `abc` in quotes and _assigned_ it to the
variable `s`.  This operation could also be written
`s <- "abc"` or `"abc" -> s`.

### Calling functions

The most common syntax for working with R was illustrated
in the string operation illustration: `paste(s, s, sep="")`.
This operation took the string variables `s` and
concatenated it with itself.  The
additional function argument `sep=""` arranges
that the concatenation is to occur without any intervening text.

The examples illustrate the use of two more functions: `rnorm`,
and `plot`.

### Packages

The family of functions and data available in an R
session depends upon the collection of packages
attached in the session.  Packages are used to
collect related functions together with their
documentation.  We can add to the collection of
functions available for plotting, and use
the ggplot approach to designing visualizations,
as follows. 
```{r dogg}
library(ggplot2)
simpdf = data.frame(x,y)
ggplot(simpdf, aes(x=x, y=y)) + geom_point()
```

### Summary: strategies for bioinformatics with R/Bioconductor

These questions should be 
considered in preparing
to work with R for bioinformatics analysis.

- What question do I want to answer with this analysis?
- What data am I going to use?  Have I clearly documented
the production of the data, including its origin, quality
assessment, data editing, sample annotations?
- What is a clear verbal description of the analysis to
be performed?
- What R/Bioconductor packages can be used to conduct
the analysis?
- If the analysis consists of a sequence of steps, should
I save interim results in case an error occurs somewhere
downstream?
- What steps will I take to ensure that the analysis is
reproducible by independent parties?

The Bioconductor software/analysis/documentation ecosystem
can help with answers to these questions.  There are many
facets to the ecosystem.  Bioconductor users form an
active community and post questions and answers at
`support.bioconductor.org`.  You can often get input
on answers to some of the questions above by browsing and
searching that site and posting your own questions.


## Basics of genome annotation: reference genomes, transcriptomes, functional annotation

### Reference genomic sequence

The `BSgenome` package defines infrastructure that supports compact
representation of genomic sequences.  We'll have a look at
a recent reference build for baker's yeast.  For this to work,
we need to know the name of the package that provides the
genomic sequence, install it, and use appropriate functions
to work with the BSgenome representation.

```{r doye1}
# BiocManager::install("BSgenome.Scerevisiae.UCSC.sacCer3") if necessary
library(BSgenome)
library(BSgenome.Scerevisiae.UCSC.sacCer3)
yeastg = BSgenome.Scerevisiae.UCSC.sacCer3
yeastg
```
Evaluation of `yeastg` in the R session produces a brief report on the
BSgenome representation.  We can look at chromosomal sequence with
```{r lkseq}
yeastg$chrI
```

What we call `yeastg` is a Bioconductor representation of the
genome of a model organism.  Similar representations are available
for more species:
```{r lkagen}
length(available.genomes())
head(available.genomes())
```

Each of the genomes listed in the output of `available_genomes()` can be obtained by
installing the associated package, attaching the package and then operating
with the eponymous object, as in `yeastg = BSgenome.Scerevisiae.UCSC.sacCer3`.
All of the objects representing the reference genomes of model organisms
are instance of a 'class'.

```{r lkcla}
class(yeastg)
```

One way of discovering the family of functions that have
been defined to work with instances of a class is to use
the `methods` function:
```{r lkme}
methods(class="BSgenome")
```

Let's try one:
```{r lkvma}
atgm = vmatchPattern("ATG", yeastg)
atgm
```
This gives us the locations of all occurrences of `ATG` in the yeast genome.

### The TxDb family

Model organism genomes and transcriptomes are catalogued at UCSC
and Ensembl.  The UCSC catalog is attached and used as follows:
```{r lktxdb}
library(TxDb.Scerevisiae.UCSC.sacCer3.sgdGene)
ytx = TxDb.Scerevisiae.UCSC.sacCer3.sgdGene
class(ytx)
ytx
```
As before, we can discover available functions for working
with such catalogues:
```{r lkcat}
methods(class="TxDb")
```
To obtain the DNA sequences of all yeast genes, use `getSeq`:
First, we need the addresses of the genes.  We use a
special representation called `GRanges`.
```{r getgadd}
y_g = genes(ytx)
y_g
```
Now we call `getSeq`:
```{r dogs}
getSeq(yeastg, y_g)
```

For human studies, the same concepts apply.
```{r lktxhum}
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb = TxDb.Hsapiens.UCSC.hg19.knownGene
transcripts(txdb)
```

### AnnotationHub example: Epigenetically defined functional annotation 

Finally, the AnnotationHub package defines a database with many
types of biological annotation.  For our last example in
this section, we consider how to obtain the ChromHmm
cell type annotation for diverse cell types.  The
AnnotationHub can be updated in real time.  We
obtain an image of the hub, and then "query".
```{r lklklk}
library(AnnotationHub)
ah = AnnotationHub()
chq = query(ah, "ChromHmm")
chq
```
The result here can be very large and so an abbreviated
view is dumped.  To learn about the cell types studied,
we use a technical call that takes advantage of knowledge
that `mcols` on a query result gives a table of features
for each resource.
```{r lkct}
mcols(chq)["AH46861",]$tags
head(table(vapply(mcols(chq)[-1,]$tags, 
   function(x)x[8],  character(1))))
```
Now we can acquire the ChromHmm labeling of genomic regions
for the H1-derived mesenchymal stem cells:
```{r lksh1}
h1msc = ah[["AH46861"]]
h1msc
table(h1msc$name)
```




## The Cancer Genome Atlas, a paradigm of federated multiomics

- TCGA, curatedTCGAData

## The OSCA book, a paradigm of computable scientific literature

- OSCA book
- probably need 1b on scRNA, one part on just the basics, on on how to work with the clusters you find
