<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>WS1 Introduction to Bioconductor 3.12 • biocwk312</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="WS1 Introduction to Bioconductor 3.12">
<meta property="og:description" content="biocwk312">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">biocwk312</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/biocwk312_agenda.html">biocwk312 workshops agenda</a>
    </li>
    <li>
      <a href="../articles/biocwk312_demo.html">biocwk312 workshops demo</a>
    </li>
    <li>
      <a href="../articles/ws1_intro.html">WS1 Introduction to Bioconductor 3.12</a>
    </li>
    <li>
      <a href="../articles/ws2_transcriptomics.html">WS2 Introduction to transcriptomics with Bioconductor 3.12</a>
    </li>
    <li>
      <a href="../articles/ws3_genetics.html">WS3 Human disease genetics with Bioconductor 3.12</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/vjcitn/biocwk312">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ws1_intro_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="ws1_intro_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="ws1_intro_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>WS1 Introduction to Bioconductor 3.12</h1>
                        <h4 class="author">Vince Carey<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>
</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/vjcitn/biocwk312/blob/master/vignettes/ws1_intro.Rmd"><code>vignettes/ws1_intro.Rmd</code></a></small>
      <div class="hidden name"><code>ws1_intro.Rmd</code></div>

    </div>

    
    
<div id="prerequisites-and-motivating-examples" class="section level1">
<h1 class="hasAnchor">
<a href="#prerequisites-and-motivating-examples" class="anchor"></a>Prerequisites and motivating examples</h1>
<div id="pre-requisites" class="section level2">
<h2 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h2>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with Rstudio</li>
<li>Basic understanding of modern genomics. For example, the distinction between whole genome sequencing and RNA-seq should be clear.</li>
<li>Basic familiarity with the concepts of statistical analysis, such as the definition of the t-test for comparing sample means, the interpretation of histograms. An understanding of experimental design is helpful.</li>
<li>Optional readings:
<ul>
<li><a href="https://cran.r-project.org/doc/manuals/r-release/R-intro.pdf">An introduction to R</a></li>
<li>
<a href="https://education.rstudio.com/learn/beginner/">The beginners page</a> at the Rstudio education portal.</li>
<li>
<a href="https://www.datacamp.com/community/tutorials/intro-bioconductor">An introduction to Bioconductor</a> at DataCamp.</li>
</ul>
</li>
</ul>
</div>
<div id="programming-and-package-use-with-r-4-0-3" class="section level2">
<h2 class="hasAnchor">
<a href="#programming-and-package-use-with-r-4-0-3" class="anchor"></a>Programming and package use with R 4.0.3</h2>
<p>R is a programming language focusing on activities important in statistical data analysis, but it is often thought of more as an interactive environment for working with data. This concept of “environment” will become clearer as we use R to perform progressively mode complex tasks.</p>
<div id="simple-illustrations" class="section level3">
<h3 class="hasAnchor">
<a href="#simple-illustrations" class="anchor"></a>Simple illustrations</h3>
<ul>
<li>simple arithmetic</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="fl">7</span><span class="op">+</span><span class="fl">5</span><span class="op">)</span><span class="op">/</span><span class="op">(</span><span class="fl">3</span><span class="op">*</span><span class="fl">6</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.6666667</span></code></pre></div>
<ul>
<li>assignment; string definitions and operations</li>
</ul>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">s</span> <span class="op">=</span> <span class="st">"abc"</span>
<span class="va">s</span>
<span class="co">#&gt; [1] "abc"</span>
<span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="va">s</span>, <span class="va">s</span>, sep<span class="op">=</span><span class="st">""</span><span class="op">)</span>
<span class="co">#&gt; [1] "abcabc"</span></code></pre></div>
<ul>
<li>random number generation</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">1234</span><span class="op">)</span> <span class="co"># setting a seed for reproducibility</span>
<span class="va">x</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>
<span class="va">x</span>
<span class="co">#&gt; [1] -1.2070657  0.2774292  1.0844412 -2.3456977  0.4291247</span></code></pre></div>
<ul>
<li>data visualization</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">y</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Normal.html">rnorm</a></span><span class="op">(</span><span class="fl">5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span></code></pre></div>
<p><img src="ws1_intro_files/figure-html/ill4-1.png" width="700"></p>
</div>
<div id="making-new-variables" class="section level3">
<h3 class="hasAnchor">
<a href="#making-new-variables" class="anchor"></a>Making new variables</h3>
<p>In the example <code>s = "abc"</code>, we formed a <em>string constant</em> by enclosing <code>abc</code> in quotes and <em>assigned</em> it to the variable <code>s</code>. This operation could also be written <code>s &lt;- "abc"</code> or <code>"abc" -&gt; s</code>.</p>
</div>
<div id="calling-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#calling-functions" class="anchor"></a>Calling functions</h3>
<p>The most common syntax for working with R was illustrated in the string operation illustration: <code><a href="https://rdrr.io/r/base/paste.html">paste(s, s, sep="")</a></code>. This operation took the string variables <code>s</code> and concatenated it with itself. The additional function argument <code>sep=""</code> arranges that the concatenation is to occur without any intervening text.</p>
<p>The examples illustrate the use of two more functions: <code>rnorm</code>, and <code>plot</code>.</p>
</div>
<div id="packages" class="section level3">
<h3 class="hasAnchor">
<a href="#packages" class="anchor"></a>Packages</h3>
<p>The family of functions and data available in an R session depends upon the collection of packages attached in the session. Packages are used to collect related functions together with their documentation. We can add to the collection of functions available for plotting, and use the ggplot approach to designing visualizations, as follows.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="va">simpdf</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">x</span>,<span class="va">y</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">simpdf</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">x</span>, y<span class="op">=</span><span class="va">y</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="ws1_intro_files/figure-html/dogg-1.png" width="700"></p>
</div>
<div id="summary-strategies-for-bioinformatics-with-rbioconductor" class="section level3">
<h3 class="hasAnchor">
<a href="#summary-strategies-for-bioinformatics-with-rbioconductor" class="anchor"></a>Summary: strategies for bioinformatics with R/Bioconductor</h3>
<p>These questions should be considered in preparing to work with R for bioinformatics analysis.</p>
<ul>
<li>What question do I want to answer with this analysis?</li>
<li>What data am I going to use? Have I clearly documented the production of the data, including its origin, quality assessment, data editing, sample annotations?</li>
<li>What is a clear verbal description of the analysis to be performed?</li>
<li>What R/Bioconductor packages can be used to conduct the analysis?</li>
<li>If the analysis consists of a sequence of steps, should I save interim results in case an error occurs somewhere downstream?</li>
<li>What steps will I take to ensure that the analysis is reproducible by independent parties?</li>
</ul>
<p>The Bioconductor software/analysis/documentation ecosystem can help with answers to these questions. There are many facets to the ecosystem. Bioconductor users form an active community and post questions and answers at <code>support.bioconductor.org</code>. You can often get input on answers to some of the questions above by browsing and searching that site and posting your own questions.</p>
</div>
</div>
<div id="basics-of-genome-annotation-reference-genomes-transcriptomes-functional-annotation" class="section level2">
<h2 class="hasAnchor">
<a href="#basics-of-genome-annotation-reference-genomes-transcriptomes-functional-annotation" class="anchor"></a>Basics of genome annotation: reference genomes, transcriptomes, functional annotation</h2>
<div id="reference-genomic-sequence" class="section level3">
<h3 class="hasAnchor">
<a href="#reference-genomic-sequence" class="anchor"></a>Reference genomic sequence</h3>
<p>The <code>BSgenome</code> package defines infrastructure that supports compact representation of genomic sequences. We’ll have a look at a recent reference build for baker’s yeast. For this to work, we need to know the name of the package that provides the genomic sequence, install it, and use appropriate functions to work with the BSgenome representation.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># BiocManager::install("BSgenome.Scerevisiae.UCSC.sacCer3") if necessary</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/BSgenome">BSgenome</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BSgenome.Scerevisiae.UCSC.sacCer3</span><span class="op">)</span>
<span class="va">yeastg</span> <span class="op">=</span> <span class="va">BSgenome.Scerevisiae.UCSC.sacCer3</span>
<span class="va">yeastg</span>
<span class="co">#&gt; Yeast genome:</span>
<span class="co">#&gt; # organism: Saccharomyces cerevisiae (Yeast)</span>
<span class="co">#&gt; # genome: sacCer3</span>
<span class="co">#&gt; # provider: UCSC</span>
<span class="co">#&gt; # release date: April 2011</span>
<span class="co">#&gt; # 17 sequences:</span>
<span class="co">#&gt; #   chrI    chrII   chrIII  chrIV   chrV    chrVI   chrVII  chrVIII chrIX  </span>
<span class="co">#&gt; #   chrX    chrXI   chrXII  chrXIII chrXIV  chrXV   chrXVI  chrM           </span>
<span class="co">#&gt; # (use 'seqnames()' to see all the sequence names, use the '$' or '[[' operator</span>
<span class="co">#&gt; # to access a given sequence)</span></code></pre></div>
<p>Evaluation of <code>yeastg</code> in the R session produces a brief report on the BSgenome representation. We can look at chromosomal sequence with</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">yeastg</span><span class="op">$</span><span class="va">chrI</span>
<span class="co">#&gt; 230218-letter DNAString object</span>
<span class="co">#&gt; seq: CCACACCACACCCACACACCCACACACCACACCACA...GGTGTGTGGGTGTGGTGTGGGTGTGGTGTGTGTGGG</span></code></pre></div>
<p>What we call <code>yeastg</code> is a Bioconductor representation of the genome of a model organism. Similar representations are available for more species:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BSgenome/man/available.genomes.html">available.genomes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] 100</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BSgenome/man/available.genomes.html">available.genomes</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; [1] "BSgenome.Alyrata.JGI.v1"                </span>
<span class="co">#&gt; [2] "BSgenome.Amellifera.BeeBase.assembly4"  </span>
<span class="co">#&gt; [3] "BSgenome.Amellifera.UCSC.apiMel2"       </span>
<span class="co">#&gt; [4] "BSgenome.Amellifera.UCSC.apiMel2.masked"</span>
<span class="co">#&gt; [5] "BSgenome.Aofficinalis.NCBI.V1"          </span>
<span class="co">#&gt; [6] "BSgenome.Athaliana.TAIR.04232008"</span></code></pre></div>
<p>Each of the genomes listed in the output of <code>available_genomes()</code> can be obtained by installing the associated package, attaching the package and then operating with the eponymous object, as in <code>yeastg = BSgenome.Scerevisiae.UCSC.sacCer3</code>. All of the objects representing the reference genomes of model organisms are instance of a ‘class’.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">yeastg</span><span class="op">)</span>
<span class="co">#&gt; [1] "BSgenome"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "BSgenome"</span></code></pre></div>
<p>One way of discovering the family of functions that have been defined to work with instances of a class is to use the <code>methods</code> function:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/methods.html">methods</a></span><span class="op">(</span>class<span class="op">=</span><span class="st">"BSgenome"</span><span class="op">)</span>
<span class="co">#&gt;  [1] [[              $               as.list         bsgenomeName   </span>
<span class="co">#&gt;  [5] coerce          commonName      countPWM        export         </span>
<span class="co">#&gt;  [9] extractAt       getSeq          injectSNPs      length         </span>
<span class="co">#&gt; [13] masknames       matchPWM        metadata        metadata&lt;-     </span>
<span class="co">#&gt; [17] mseqnames       names           organism        provider       </span>
<span class="co">#&gt; [21] providerVersion releaseDate     releaseName     seqinfo        </span>
<span class="co">#&gt; [25] seqinfo&lt;-       seqnames        seqnames&lt;-      show           </span>
<span class="co">#&gt; [29] snpcount        SNPlocs_pkgname snplocs         sourceUrl      </span>
<span class="co">#&gt; [33] vcountPattern   vcountPDict     Views           vmatchPattern  </span>
<span class="co">#&gt; [37] vmatchPDict    </span>
<span class="co">#&gt; see '?methods' for accessing help and source code</span></code></pre></div>
<p>Let’s try one:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atgm</span> <span class="op">=</span> <span class="fu">vmatchPattern</span><span class="op">(</span><span class="st">"ATG"</span>, <span class="va">yeastg</span><span class="op">)</span>
<span class="va">atgm</span>
<span class="co">#&gt; GRanges object with 444764 ranges and 0 metadata columns:</span>
<span class="co">#&gt;            seqnames      ranges strand</span>
<span class="co">#&gt;               &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;        [1]     chrI     283-285      +</span>
<span class="co">#&gt;        [2]     chrI     335-337      +</span>
<span class="co">#&gt;        [3]     chrI     388-390      +</span>
<span class="co">#&gt;        [4]     chrI     436-438      +</span>
<span class="co">#&gt;        [5]     chrI     492-494      +</span>
<span class="co">#&gt;        ...      ...         ...    ...</span>
<span class="co">#&gt;   [444760]     chrM 85069-85071      -</span>
<span class="co">#&gt;   [444761]     chrM 85322-85324      -</span>
<span class="co">#&gt;   [444762]     chrM 85470-85472      -</span>
<span class="co">#&gt;   [444763]     chrM 85681-85683      -</span>
<span class="co">#&gt;   [444764]     chrM 85776-85778      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 17 sequences from an unspecified genome</span></code></pre></div>
<p>This gives us the locations of all occurrences of <code>ATG</code> in the yeast genome.</p>
</div>
<div id="cds-genes-transcripts" class="section level3">
<h3 class="hasAnchor">
<a href="#cds-genes-transcripts" class="anchor"></a>CDS, Genes, transcripts</h3>
<p>Model organism genomes and transcriptomes are catalogued at UCSC and Ensembl. The UCSC catalog is attached and used as follows:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">TxDb.Scerevisiae.UCSC.sacCer3.sgdGene</span><span class="op">)</span>
<span class="va">ytx</span> <span class="op">=</span> <span class="va">TxDb.Scerevisiae.UCSC.sacCer3.sgdGene</span>
<span class="fu"><a href="https://rdrr.io/r/base/class.html">class</a></span><span class="op">(</span><span class="va">ytx</span><span class="op">)</span>
<span class="co">#&gt; [1] "TxDb"</span>
<span class="co">#&gt; attr(,"package")</span>
<span class="co">#&gt; [1] "GenomicFeatures"</span>
<span class="va">ytx</span>
<span class="co">#&gt; TxDb object:</span>
<span class="co">#&gt; # Db type: TxDb</span>
<span class="co">#&gt; # Supporting package: GenomicFeatures</span>
<span class="co">#&gt; # Data source: UCSC</span>
<span class="co">#&gt; # Genome: sacCer3</span>
<span class="co">#&gt; # Organism: Saccharomyces cerevisiae</span>
<span class="co">#&gt; # Taxonomy ID: 4932</span>
<span class="co">#&gt; # UCSC Table: sgdGene</span>
<span class="co">#&gt; # Resource URL: http://genome.ucsc.edu/</span>
<span class="co">#&gt; # Type of Gene ID: Name of canonical transcript in cluster</span>
<span class="co">#&gt; # Full dataset: yes</span>
<span class="co">#&gt; # miRBase build ID: NA</span>
<span class="co">#&gt; # transcript_nrow: 6692</span>
<span class="co">#&gt; # exon_nrow: 7034</span>
<span class="co">#&gt; # cds_nrow: 7034</span>
<span class="co">#&gt; # Db created by: GenomicFeatures package from Bioconductor</span>
<span class="co">#&gt; # Creation time: 2015-10-07 18:20:42 +0000 (Wed, 07 Oct 2015)</span>
<span class="co">#&gt; # GenomicFeatures version at creation time: 1.21.30</span>
<span class="co">#&gt; # RSQLite version at creation time: 1.0.0</span>
<span class="co">#&gt; # DBSCHEMAVERSION: 1.1</span></code></pre></div>
<p>As before, we can discover available functions for working with such catalogues:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/methods.html">methods</a></span><span class="op">(</span>class<span class="op">=</span><span class="st">"TxDb"</span><span class="op">)</span>
<span class="co">#&gt;  [1] $                      $&lt;-                    annotatedDataFrameFrom</span>
<span class="co">#&gt;  [4] as.list                asBED                  asGFF                 </span>
<span class="co">#&gt;  [7] assayData              assayData&lt;-            cds                   </span>
<span class="co">#&gt; [10] cdsBy                  cdsByOverlaps          coerce                </span>
<span class="co">#&gt; [13] columns                combine                contents              </span>
<span class="co">#&gt; [16] dbconn                 dbfile                 dbInfo                </span>
<span class="co">#&gt; [19] dbmeta                 dbschema               disjointExons         </span>
<span class="co">#&gt; [22] distance               exons                  exonsBy               </span>
<span class="co">#&gt; [25] exonsByOverlaps        ExpressionSet          extractUpstreamSeqs   </span>
<span class="co">#&gt; [28] featureNames           featureNames&lt;-         fiveUTRsByTranscript  </span>
<span class="co">#&gt; [31] genes                  initialize             intronsByTranscript   </span>
<span class="co">#&gt; [34] isActiveSeq            isActiveSeq&lt;-          isNA                  </span>
<span class="co">#&gt; [37] keys                   keytypes               mapIds                </span>
<span class="co">#&gt; [40] mapIdsToRanges         mappedkeys             mapRangesToIds        </span>
<span class="co">#&gt; [43] mapToTranscripts       metadata               microRNAs             </span>
<span class="co">#&gt; [46] nhit                   organism               promoters             </span>
<span class="co">#&gt; [49] revmap                 sample                 sampleNames           </span>
<span class="co">#&gt; [52] sampleNames&lt;-          saveDb                 select                </span>
<span class="co">#&gt; [55] seqinfo                seqinfo&lt;-              seqlevels&lt;-           </span>
<span class="co">#&gt; [58] seqlevels0             show                   species               </span>
<span class="co">#&gt; [61] storageMode            storageMode&lt;-          taxonomyId            </span>
<span class="co">#&gt; [64] threeUTRsByTranscript  transcripts            transcriptsBy         </span>
<span class="co">#&gt; [67] transcriptsByOverlaps  tRNAs                  updateObject          </span>
<span class="co">#&gt; see '?methods' for accessing help and source code</span></code></pre></div>
<ul>
<li>BSgenome, EnsDb, AnnotationHub</li>
</ul>
</div>
</div>
<div id="the-cancer-genome-atlas-a-paradigm-of-federated-multiomics" class="section level2">
<h2 class="hasAnchor">
<a href="#the-cancer-genome-atlas-a-paradigm-of-federated-multiomics" class="anchor"></a>The Cancer Genome Atlas, a paradigm of federated multiomics</h2>
<ul>
<li>TCGA, curatedTCGAData</li>
</ul>
</div>
<div id="the-osca-book-a-paradigm-of-computable-scientific-literature" class="section level2">
<h2 class="hasAnchor">
<a href="#the-osca-book-a-paradigm-of-computable-scientific-literature" class="anchor"></a>The OSCA book, a paradigm of computable scientific literature</h2>
<ul>
<li>OSCA book</li>
<li>probably need 1b on scRNA, one part on just the basics, on on how to work with the clusters you find</li>
</ul>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p><a href="mailto:stvjc@channing.harvard.edu" class="email">stvjc@channing.harvard.edu</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Vince Carey.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
